### PR TITLE
Overhaul the infrastructure

### PR DESCRIPTION
**Pull request by jnyrup**
This PR brings FluentAssertions.Json all the love that FluentAssertions have had in terms of editorconfig, analyzers, code coverage, etc.

The PR is split up into a commit per feature.

TODO: 
* Add `NUGETAPIKEY` to https://github.com/fluentassertions/fluentassertions.json/settings/secrets/actions
* Disable AppVeyor

**6.0.0 from nuget.org**
image

**this PR**
image

### CODE REVIEW

File: Tests/Approval.Tests/ApiApproval.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace Approval.Tests
+{
+    [UsesVerify]
+    public class ApiApproval
+    {
+        [Theory]
+        [InlineData("net47")]
+        // TODO: [InlineData("netstandard2.0")] throws AssemblyResolutionException for Jonas
+        public Task ApproveApi(string frameworkVersion)
```
Comment (dennisdoomen) [inline Tests/Approval.Tests/ApiApproval.cs:23]: ⛏️ (nitpick) This is a particularly ugly block of code.

File: Src/FluentAssertions.Json/JTokenDifferentiator.cs [hunk 2]
```diff
@@ -293,47 +269,39 @@ public Difference(DifferenceKind kind, JPath path)
...
         private JPath Path { get; }
+
         private object Actual { get; }
+
         private object Expected { get; }
 
         public override string ToString()
         {
-            switch (Kind)
+            return Kind switch
```
Comment (dennisdoomen) [inline Src/FluentAssertions.Json/JTokenDifferentiator.cs:282]: 👍🏻 _If_ you need a `switch` statement, this is indeed the best way to represent it.

File: Src/FluentAssertions.Json/FluentAssertions.Json.csproj [hunk 3]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <TargetFrameworks>net47;netstandard2.0</TargetFrameworks>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <NoWarn>1701;1702;1705;1591;1574;1572;1573;419</NoWarn>
+    <NoWarn>$(NoWarn);1701;1702;1705;1591;1574;1572;1573;419</NoWarn>
```
Comment (dennisdoomen) [inline Src/FluentAssertions.Json/FluentAssertions.Json.csproj:6]: ❓ Are you planning to squash this commit in the previous one?

**Review by dennisdoomen — APPROVED** [review]
What a beautiful piece of work. I was thinking the exact thing this morning, but lucky for me, you took it upon yourself.

File: Tests/Approval.Tests/ApiApproval.cs (hunk 1 above)
Comment (jnyrup) [reply Tests/Approval.Tests/ApiApproval.cs:23]: Is there a concrete change you'd like or is this just an observation?

Comment (dennisdoomen) [reply Tests/Approval.Tests/ApiApproval.cs:23]: Only if you have time, hence I marked it as a nitpick.

Comment (jnyrup) [reply Tests/Approval.Tests/ApiApproval.cs:23]: Alright.
Then I'll defer the prettifying to whoever gets to it first

**Comment by jnyrup** [comment]
Hmm... shouldn't the addition of `.github/workflows/build.yml` have triggered a build?

**Pull request merged**